osu.Server.Spectator/Hubs/MultiplayerHub.cs

```diff
@@ -50,7 +50,7 @@ public async Task<MultiplayerRoom> JoinRoom(long roomId)
                 if (userUsage.Item != null)
                 {
                     // if the user already has a state, it means they are already in a room and can't join another without first leaving.
-                    throw new InvalidStateException("Can't join a room when already in another room.");
+                    throw new InvalidStateException($"Can't join room {roomId} when already in room {userUsage.Item.CurrentRoomID}.");
```

Comment:
Sure we don't want a `LogAndThrow()` sort of thing for these? These won't log the current user context ID, or will they?

Some throws already include user IDs within exception messages - might supersede those with that method if it is added.

Also as a side thought, we're exposing these to the users (as any errors in multi operations are just marshaled as plain server-side exception through the wire), so we might want something that's more structured, like a user-facing message along with a context object that gets logged to console on top of that. Maybe going too far with that one, I don't know. Not sure how permanent this logging is intended to be.

Reply:
Invalid state are client-side issues, so this should probably just be reverted. As you say, this should be a user parsable for now.